Sources/Fuzzilli/Mutators/TypeAwareInputMutator.swift

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// A version of the InputMutator that takes type information into account, and so can be more aggressive
+public class TypeAwareInputMutator: BaseInstructionMutator {
+    public init() {
+        super.init(maxSimultaneousMutations: defaultMaxSimultaneousMutations * 2)
+    }
+    
+    public override func canMutate(_ instr: Instruction) -> Bool {
+        return instr.numInputs > 0
+    }
+    
+    public override func mutate(_ instr: Instruction, _ b: ProgramBuilder) {
+        var inouts = b.adopt(instr.inouts)
+
+        // Replace one input
+        let selectedInput = Int.random(in: 0..<instr.numInputs)
+        let type = b.type(of: inouts[selectedInput]).generalize()
+        b.trace("Mutating input \(selectedInput) of generalized type \(type)")
+        // Inputs to block end instructions must be taken from the outer scope since the scope closed by the instruction is currently still active.
+        let replacement = b.randVar(ofType: type, excludeInnermostScope: instr.isBlockEnd) ?? b.randVar(excludeInnermostScope: instr.isBlockEnd)
```

Comment:
Since we default to picking a random var if we can't find one of a desired type, Is there a reason we can't merge `TypeAwareInputMutator` and `InputMutator`?

Reply:
Good point, thanks! So I've now instead added a parameter to the existing InputMutator, which I think is nicer. I'd like the two to be separate though, so they can be weighted differently.

Reply:
Yep, that sounds good to me